lightning/src/ln/functional_test_utils.rs

```diff
@@ -693,6 +693,61 @@ pub fn create_announced_chan_between_nodes_with_value<'a, 'b, 'c, 'd>(nodes: &'a
 	(chan_announcement.1, chan_announcement.2, chan_announcement.3, chan_announcement.4)
 }
 
+pub fn create_unannounced_chan_between_nodes_with_value<'a, 'b, 'c, 'd>(nodes: &'a Vec<Node<'b, 'c, 'd>>, a: usize, b: usize, channel_value: u64, push_msat: u64, a_flags: InitFeatures, b_flags: InitFeatures) -> (msgs::FundingLocked, Transaction) {
+	let mut no_announce_cfg = test_default_channel_config();
+	no_announce_cfg.channel_options.announced_channel = false;
```

Comment:
Hmm, all the existing utilities end up either calling `create_channel` itself or expecting to exchange `AnnouncementSignatures` at the end. I could rejigger the utilities above to separate those bits out, if you prefer? I don't really have a strong feeling between that vs having some code duplication in utilities right around each other.

Reply:
Don't feel too strongly. I think I already asked another contributor to do the same in another PR, heh.

Reply:
Hmm, on second thought I'm kinda struggling to see how to do this - the top 7 lines could be split out into three lines and four lines, but that doesn't seem so useful. Then there's the funding generation, which almost can be combined, except that we rely on exchanging the funding_locked messages in order so that we only get one `channel_update` out at a time - we could swap for the `create_chan_between_nodes_with_value_confirm_first` helper, but then we're stick peeling apart multiple pending messages which takes more LoC than we save :(.

Reply:
No problem with passing on this. My concern was more with code repetition resulting in needing to modify two places rather than reducing lines of code.

Reply:
Yea, I agree, its unclear how to resolve it in this case, though, without digging through a number of other tests and changing their util method calls, which is yet more diff in this PR :/